packages/flutter_tools/lib/src/template.dart

```diff
@@ -252,12 +262,22 @@ class Template {
 
       // Step 5: This file does not end in .tmpl but is in a directory that
       //         does. Directly copy the file to the destination.
-
+      _validateReadPermissions(sourceFile);
       sourceFile.copySync(finalDestinationFile.path);
     });
 
     return fileCount;
   }
+
+  /// Attempt to read a byte from the file to ensure that read permissions are correct.
+  ///
+  /// If this fails with a certain error code, the [ErrorHandlingFileSystem] will
+  /// trigger a tool exit with a better message.
+  void _validateReadPermissions(File file) {
+    final RandomAccessFile randomAccessFile = file.openSync();
```

Comment:
It doesn't look like `ErrorHandlingFile` overrides `openSync()`.

Reply:
Ahh, yeah I did not run my integration test with this change after switching to the error handling file system. I could modify this so that it skips the random access file.